### ISSUE TITLE
Please keep updating this project

### ISSUE THREAD

**Issue by glli** [Xtrendence/X-Notes#2]
I compared X-Notes with 10+ other [php note-taking projects](https://github.com/topics/note-taking?l=php) and find this one is the most well-designed note with super useful sharing and lock functions. The requirements and deployment are very simple (no database needed) however the notes are very safe. I really like X-Notes and decide to deploy it to my personal server.
The only shortcoming is the note editing. As a wysiwyg editor, X-Notes looks pretty embarrassed with only a few function limited buttons (though very useful). It can definitely be improved like [this one](https://github.com/pollmix/notepad):
![image](https://user-images.githubusercontent.com/3065472/134630216-b37388bb-ce5f-40bd-bbeb-e64527e631c3.png)
Anyway I really appreciate author's work. This is a brilliant project!

**Comment by Ry3yr** [comment]
That is ok.
You can copy & paste from other wysiwyg editors and they'll work.

I tossed my nimnbusnotes one in and they display perfectly.

Awesome php/js app.
Finally free of the money grub shackles (hi evernote...)

If you still see the need, fork this X-notes and add what you need to the existing bar.
I have this one on my webpage f.e.:
https://ry3yr.github.io/OSTR/Diarykeepers_Homepage/HTMLEditor.html
These features should be implementable

**Comment by nzpcguru** [comment]
is there a way to have more than one user ??

**Comment by Ry3yr** [comment]
I̶n̶ t̶h̶e̶ s̶a̶m̶e̶ x̶n̶o̶t̶e̶s̶ ?̶ 

 h̶m̶m̶. d̶e̶f̶a̶u̶l̶t̶ i̶s̶ a̶d̶m̶i̶n̶.  

y̶o̶u̶ c̶o̶u̶l̶d̶ h̶a̶v̶e̶ m̶u̶l̶t̶i̶p̶l̶e̶ x̶n̶o̶t̶e̶s̶ f̶o̶l̶d̶e̶r̶s̶. (̶E̶a̶c̶h̶ o̶f̶ t̶h̶e̶m̶ h̶a̶v̶i̶n̶g̶ "a̶d̶m̶i̶n̶" a̶s̶ u̶s̶e̶r̶ o̶f̶c̶)̶  

x̶n̶o̶t̶e̶s̶/s̶r̶c̶/c̶f̶g̶/a̶c̶c̶o̶u̶n̶t̶.c̶o̶n̶f̶i̶g̶ c̶o̶n̶t̶a̶i̶n̶s̶ t̶h̶e̶ u̶s̶e̶r̶s̶.

Edit:
Yes, in the SETTINGS, you can add another user :)
![Screenshot_2024-03-11-01-07-13-70_0311c9f6806a66343c45622522faa000](https://github.com/Xtrendence/X-Notes/assets/17431197/bddd4ba0-1571-44dc-8ccb-187e2dd11eb7)

**Comment by Xtrendence** [comment]
Hey guys :) 

First of all, thank you for the kind words and for supporting the project. At the time I made this, I was quite new to development and around my first year in uni if I remember right. I didn't really have much time to revisit old projects and such, so I didn't really go anywhere with this. 

Since then, I work full time as a dev, but have a clear schedule and such, and have started developing personal projects again. This one gets the odd email and such here and there asking me to update it, so I think I'll start here. I'll branch off this (not fork), and eventually merge it back to keep the stars and name of the repo. 

I will add more formatting features and multiple user support. @Ry3yr is correct in pointing out that if you have multiple `xnotes` folders, then you could efffectively have multiple users and use the settings page to change the username (keep in mind, account.cfg is overwritten when you change username or password, not appended to, and the code is written to only check 1 username/password in there). So yeah, multiple `xnotes` would work, but changing username won't add a new user, just replace the current one. 

The question I have for you guys is whether you'd be okay with X:/Notes having a Node.js API, as that's what I'm infinitely more experienced in nowadays, or is PHP is an important part of it? Keep in mind I'd likely use something like Symfony or Laravel nowadays anyway both to learn something new and to tackle PHP's issues. Obviously, in either case, I'd provide full step-by-step setup instructions.

**Comment by Ry3yr** [comment]
Personally I am verily against Node js and prefer PHP as that gurantees drag and drop for most things.
(I just tossed this onto my server, it even runs on android via https://apkcombo.com/de/palapa-web-server/com.alfanla.android.pws/download/apk#urusai.social/@alcea/111761622206017944#i.ibb)

NodeJS is a relatively new technology and while I see it having a use in many projects (Puppeteer), I think Xnotes is fine in staying php :P

TL;DR:
I cannot see what NodeJS could help XNotes with.
PHP is longstanding.
I trust it.

Adding another requirement makes software harder to use, increasing the entrybarrier.
I'd personally try and keep things as simple as possible to ease use for the enduser :)

**Comment by glli** [comment]
@Xtrendence 

I just checked this thread today due to multiple user needs and saw your reply. I prefer using pure PHP because of its simplicity, especially since I'm deploying it on my DD-WRT router, which has very limited resources but handles PHP scripts well. Scalability isn't a concern for me, as only a few family members use my service, so I don't need the advantages that Node provides.

Thank you for coming back to work on this project again and look forward to any updates from it.

**Comment by glli** [comment]
@nzpcguru @Ry3yr  I just updated the repo a little bit to support multiple users for my family members, you can view it here:
https://github.com/glli/X-Notes

**Comment by Xtrendence** [comment]
> @nzpcguru @Ry3yr  I just updated the repo a little bit to support multiple users for my family members, you can view it here:
> https://github.com/glli/X-Notes

Thank you for doing this! Unfortunately my time to work on personal projects is much more limited than I thought, so it's awesome to see someone else expand upon this project.

**Comment by Ry3yr** [comment]
Personally I had issues with corruption after exceeding a certain threshold of message body for singular notes
(My notes are really expansive)

Tailing and loosing bottom document stuff.

I'm not sure how reliable xnote works agains file corruption.

Good to see some happenings tho.

Nimbusnotes/Fusebase and Evernote are far too paywalled nowadays
